### ISSUE TITLE
2021.1.1: parallel_for_each fails to compile for noncopyable elements

### ISSUE THREAD

**Issue by Lastique** [oneapi-src/oneTBB#321]
The following test case fails to compile with oneTBB 2021.1.1 and gcc 10.2:

```
#include <tbb/parallel_for_each.h>

struct my_element
{
    my_element(unsigned int n) : m_n(n) {}
    my_element(my_element const&) = delete;
    my_element& operator= (my_element const&) = delete;

    unsigned int m_n;
};

int main()
{
    my_element elems[2] = { {0}, {1} };
    tbb::parallel_for_each(elems, elems + 2, [](my_element& elem) { ++elem.m_n; });
}
```

```
$ g++ -std=gnu++17 -I include -o parallel_for_each_noncopyable.o -c parallel_for_each_noncopyable.cpp
In file included from include/tbb/parallel_for_each.h:17,
                 from parallel_for_each_noncopyable.cpp:1:
include/tbb/../oneapi/tbb/parallel_for_each.h: In instantiation of ‘tbb::detail::d1::feeder_item_task<Body, Item>::feeder_item_task(ItemType&&, tbb::detail::d1::feeder_item_task<Body, Item>::feeder_type&, tbb::detail::d1::small_object_allocator&) [with ItemType = my_element; Body = main()::<lambda(my_element&)>; Item = my_element; tbb::detail::d1::feeder_item_task<Body, Item>::feeder_type = tbb::detail::d1::feeder_impl<main()::<lambda(my_element&)>, my_element>]’:
include/tbb/../oneapi/tbb/detail/_small_object_pool.h:63:35:   required from ‘Type* tbb::detail::d1::small_object_allocator::new_object(Args&& ...) [with Type = tbb::detail::d1::feeder_item_task<main()::<lambda(my_element&)>, my_element>; Args = {my_element, tbb::detail::d1::feeder_impl<main()::<lambda(my_element&)>, my_element>&, tbb::detail::d1::small_object_allocator&}]’
include/tbb/../oneapi/tbb/parallel_for_each.h:174:50:   required from ‘void tbb::detail::d1::feeder_impl<Body, Item>::internal_add_move(Item&&) [with Body = main()::<lambda(my_element&)>; Item = my_element]’
include/tbb/../oneapi/tbb/parallel_for_each.h:171:10:   required from here
include/tbb/../oneapi/tbb/parallel_for_each.h:125:27: error: use of deleted function ‘my_element::my_element(const my_element&)’
  125 |         my_allocator(alloc)
      |                           ^
parallel_for_each_noncopyable.cpp:6:5: note: declared here
    6 |     my_element(my_element const&) = delete;
      |     ^~~~~~~~~~
```

The code compiled with previous TBB releases.

**Comment by alexey-katranov** [comment]
Notify: @ivankochin , @kboyarinov

**Comment by kboyarinov** [comment]
Hi, @Lastique,

Compilation issues for parallel_for_each algorithm for specific value types were fixed in [oneTBB 2021.2.0](https://github.com/oneapi-src/oneTBB/releases/tag/v2021.2.0)
If you still have any issues, feel free to contact us.
